app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsServiceNew.scala

```diff
@@ -139,11 +166,58 @@ class NrsServiceNew @Inject() (
         )
     }
 
-  private def retrieveUserAuthToken(hc: HeaderCarrier): String =
+  private def retrieveUserAuthToken()(implicit hc: HeaderCarrier): String =
     hc.authorization match {
       case Some(Authorization(authToken)) => authToken
       case _                              =>
         logger.warn("[NrsService] - No auth token available for NRS")
         throw new InternalServerException("No auth token available for NRS")
     }
+
+  def processAllWithLock()(implicit hc: HeaderCarrier): Future[Done] =
+    lockService
+      .withLock {
+        processAll()
+      }
+      .map {
+        _.getOrElse {
+          logger.info(
+            s"Could not acquire lock on nrsWorkItemRepository for thing"
+          )
+          Done
+        }
+      }
```

Comment:
Just to make sure I understand the intention - we're locking *ANY* nrs submission to one service? Why can't we have multiple deployments handling NRS submissions?